src/DynamoUtilities/PIIDetector.cs

```diff
@@ -0,0 +1,125 @@
+
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Dynamo.Utilities
+{
+    /// <summary>
+    /// Helper Class for removing PII Data from a JSON workspace
+    /// </summary>
+    internal static class PIIDetector
+    {
+        /// <summary>
+        /// Removes the PII data from a JSON workspace
+        /// </summary>
+        /// <param name="jsonObject"></param>
+        /// <returns></returns>
+        public static JObject RemovePIIData(JObject jsonObject)
```

Comment:
@mjkkirschner I have applied your suggestions, the function will be used to get a serialized workspace without PII data (https://jira.autodesk.com/browse/DYN-5964), based on a well formatted and confident json object, it will be called when an user close a workspace / graph (potentially) after checking if it's valid for send to Forge. Yes, since the project doesn't have reference to the types, the comparisons are performed to explicit values.